shockMongoRecordsToS3.py

```diff
@@ -0,0 +1,178 @@
+#!/usr/bin/env python3
+
+'''
+This script converts a subset of Shock (https://github.com/kbase/Shock) node records to
+BlobStore (https://github.com/kbase/blobstore) records.
+The script does not alter the Shock backend records and may be re-run multiple times without issue.
+
+The script determines which files to transfer by looking in the S3 bucket that contains the
+converted Shock files. It expects key names based on a uuid in the same way Shock directories are
+based on the UUID; for example the UUID 06f5d3ec-8ebf-4d32-8c1c-41e27e40b7fd would be
+
+06/f5/d3/06f5d3ec-8ebf-4d32-8c1c-41e27e40b7fd
+
+... as a key in S3 or a directory tree in Shock.
+
+To use:
+1) Start the blobstore at least once so that the proper indexes are created in MongoDB.
+2) Fill in the configuration variables below and run the script normally.
+'''
+
+###### CONFIGURATION VARIABLES ######
+
+CONFIG_MONGO_SHOCK_HOST = 'localhost'
+CONFIG_MONGO_SHOCK_USER = ''
+CONFIG_MONGO_SHOCK_PWD = ''
+CONFIG_MONGO_SHOCK_DATABASE = 'ShockDB'
+
+CONFIG_MONGO_BLOBSTORE_HOST = 'localhost'
+CONFIG_MONGO_BLOBSTORE_USER = ''
+CONFIG_MONGO_BLOBSTORE_PWD = ''
+CONFIG_MONGO_BLOBSTORE_DATABASE = 'bs_test'
+
+CONFIG_S3_HOST = 'http://localhost:9000'
+# The bucket name must obey https://docs.aws.amazon.com/AmazonS3/latest/dev/BucketRestrictions.html
+# with the extra restriction that periods are not allowed.
+CONFIG_S3_BUCKET = 'blobstore'
+CONFIG_S3_ACCESS_KEY = 'access key goes here'
+CONFIG_S3_ACCESS_SECRET = 'access secret goes here'
+CONFIG_S3_REGION = 'us-west-1'
+
+#### END CONFIGURATION VARIABLES ####
+
+import boto3
```

Comment:
Kind weird to have imports in between a bunch of declarations, but no biggie.

Reply:
I just wanted to have the user configurable params at the top and separate them from the constants.